Extensions/TileMap/collision/TransformedTileMap.ts

```diff
@@ -701,22 +733,110 @@ namespace gdjs {
+        const tileMap = this.layer.tileMap;
+        const width = tileMap.getTileWidth();
+        const height = tileMap.getTileHeight();
+
+        // Extend the hit boxes.
+        // It avoids small objects to be pushed side way into a wall when they
+        // should be pop out of the wall.
+
+        const hasFullHitBox =
+          definitionHitboxes.length === 1 && definition.hasFullHitBox(tag);
+        if (hasFullHitBox) {
+          const isLeftFull = this._hasNeighborFullHitBox(-1, 0);
+          const isRightFull = this._hasNeighborFullHitBox(1, 0);
+          const isTopFull = this._hasNeighborFullHitBox(0, -1);
+          const isBottomFull = this._hasNeighborFullHitBox(0, 1);
+
+          let hitBoxesCount = 0;
+          if (isLeftFull || isRightFull) {
+            let minX = isLeftFull ? -width : 0;
+            let maxX = isRightFull ? 2 * width : width;
+            if (this.hitBoxes.length === 0) {
+              // TODO Avoid to allocate a Polygon
+              this.hitBoxes[0] = gdjs.Polygon.createRectangle(0, 0);
+            }
+            TransformedCollisionTile.setRectangle(
+              this.hitBoxes[0],
+              minX,
+              0,
+              maxX,
+              height
+            );
+            hitBoxesCount++;
+          }
+          if (isTopFull || isBottomFull) {
+            let minY = isTopFull ? -height : 0;
+            let maxY = isBottomFull ? 2 * height : height;
+            if (this.hitBoxes.length < hitBoxesCount + 1) {
+              // TODO Avoid to allocate a Polygon
+              this.hitBoxes[hitBoxesCount] = gdjs.Polygon.createRectangle(0, 0);
+            }
+            TransformedCollisionTile.setRectangle(
+              this.hitBoxes[hitBoxesCount],
+              0,
+              minY,
+              width,
+              maxY
+            );
+            hitBoxesCount++;
+          }
+          if (hitBoxesCount === 0) {
+            if (this.hitBoxes.length === 0) {
+              // TODO Avoid to allocate a Polygon
+              this.hitBoxes[0] = gdjs.Polygon.createRectangle(0, 0);
+            }
+            TransformedCollisionTile.setRectangle(
+              this.hitBoxes[0],
+              0,
+              0,
+              width,
+              height
+            );
+            hitBoxesCount++;
+          }
+          this.hitBoxes.length = hitBoxesCount;
+        } else {
+          if (this.hitBoxes.length === 0) {
+            // This can't happen in practice as only the simple tile map can be
+            // modify and it only contains full hit boxes.
+            this.hitBoxes[0] = gdjs.Polygon.createRectangle(0, 0);
+          }
+          for (
+            let polygonIndex = 0;
+            polygonIndex < definitionHitboxes.length;
+            polygonIndex++
+          ) {
+            const defPolygon = definitionHitboxes[polygonIndex];
+            const polygon = this.hitBoxes[polygonIndex];
```

Comment:
Is there a risk we're going out of bounds (polygonIndex >= 1, and this.hitBoxes is just a single rectangle?)?